### ISSUE TITLE
Ontology

### ISSUE THREAD

**Issue by WayneEDick** [w3c/low-vision-a11y-tf#3]
<body>

<h1>  Ontology of Low Vision Web Accessibility</h1>

<p>This document describes the top level of an ontology designed to record, discover and document the user needs, causes and IT requirements for web users with low vision. It focuses on the functional-limits of individuals diagnosed with low vision. From these functional limits we attempt to infer barriers to access to information technology. This can be used by the Low Vision Task Force to devise, document and defend recommendations for extensions to WCAG 2.0. In addition, the ontology can be used as a resource for developing informative support documents and sufficient techniques for conformance.
</p>

<h2> Top Level Base Classes:
</h2>

<p>There are four basic entry points to the ontology: Functional limits&mdash; the ways visual performance are restricted, Diagnosis&mdash; a named cause of low vision, Physical conditions&mdash; physical and behavioral traits that accompany a  diagnosis,  and Information Technology Barriers&mdash; IT configurations  that restrict access to documents for individuals with low vision.
</p>

<h3> Functional Limits (Class Name: functional-limits)
</h3>

<p>Functional limits break into: Color vision, Contrast sensitivity, Discomfort, Fatigue, Field Loss, Fixation, Interaction / Exacerbation of other limits, Loss of detail, Light Sensitivity, Impaired Perception / Confusion, Rigidity, Reading difficulty, Tracking / Scanning, Visual acuity, Visual distortion, Visual Fluctuation, Visual-motor
</p>

<h3> Physical conditions (Class Name: physical-condition)
</h3>

<p>The following physical conditions are critical causes of limits:</p>

<p> astigmatism&mdash; variable focus like a skewing, cataract&mdash; cloudy obstruction in the lens, corneal damage&mdash; any obstruction to the clear structure that protects the lens, glaucoma&mdash; intra ocular pressure resulting to retinal and other damage, iritis&mdash; inflammation of the iris, keratoconus&mdash; malformation of the cornea causing blurring, myopia&mdash; near sightedness, nystagmus&mdash; involuntary eye movement, photophobia&mdash; extreme sensitivity to light and glare, ptosis&mdash; involuntary closing of the eye, refractive error&mdash; focus length of the lens does not match the length of the eye, retinal detachment&mdash; critical condition in which the retina (the visual receptors)  pulls away from the blood layer, retinal damage&mdash; a broad range of injuries to the retina which contains the eye&mdash;s visual receptors, scotomas&mdash; permanent blind or blurry spots in the visual field, strabismus&mdash; imbalance in the ocular muscles, and vitreous obstruction&mdash; floaters or other opaque or translucent patches in the clear jell filling the center of the eye.
</p>

<h3> Diagnosis: (Class Name: diagnosis)
</h3>

<p>A physical condition can be a diagnosis, but not all conditions are diagnoses. Also, not all diagnoses are physical conditions. Every physical condition is a consequence of some diagnosis. This list diagnoses below will be linked to definitions in a glossary. The following are diagnoses.
</p>

<p>Achromotopsia , Albinism, Amblyopia , Aniridia , Anophthalmia, Aphakia, Astigmatism, Buphthalmos , Cataracts, Chorioretinitis, Coats' Disease, Coloboma, Color deficiency -See Achromotopsia, Cone Monochromacy -See Achromotopsia, Corneal Ulcers, Corneal Opacities, Corneal Scarring, Keratitis, Interstitial Keratitis, Cortical Visual Impairment ,  Diabetic Retinopathy, Diplopia, Dislocated Lens, Enucleation, (Esophoria, Esotropia, Exophoria,  Exotropia) - See strabismus, Glaucoma, Hemianopia , Histoplasmosis , (Hyperopia (Farsightedness), Hyperphoria, Hypertropia, Hypophoria, Hypotropia) - See strabismus, Hypoplasia&mdash; See Optic Atrophy, Keratitis See Corneal Ulcer, Keratoconus (KC), Leber's Congenital Amaurosis, Leber's Hereditary Optic Neuropathy (LHON, Leber's Optic Atrophy),  Macular Degeneration , Microphthalmia (Microphthalmos, nanophthalmia, nanophthalmos),  Muscle Imbalance See Strabismus and Amblyopia, Myopia (Simple and Degenerative Myopia, nearsightedness),  Nystagmus, Optic Atrophy (Optic Nerve Atrophy), Optic Nerve Hypoplasia(ONH), Peter's Anomaly, Photophobia, Phthisis bulbi, Presbyopia, Ptosis, Retinal Detachment, Retinal Dysplasia, Retinitis Pigmentosa (RP), Retinoblastoma, Retinopathy of Prematurity (ROP, retrolental Fibroplasia), Retinoschisis, Rod Achromacy - See Achromotopsia, Scotoma (pl. scotomata, scotomas), Septo-Optic Dysplasia See Optic Nerve Hypoplasia and "Syndromes", Strabismus (muscle imbalance), Toxoplasmosis, Trachoma, Uveitis, Wilm's Tumor
</p>

<h3> IT Barriers: (Class Name: barriers)
</h3>

<p>Barrier classes: Legibility (P)&mdash; the ability to recognize and group letters into words, Readability (O and U)&mdash;  the ability to perceive and comprehend entire written documents , Layout and Viewport Capacity (O and U)&mdash; the visual organization of document elements on the viewport given the amount of usable space on a viewport at a level of enlargement that  ensures legibility and readability of all written material, Way Finding (O and U)&mdash; locating objects within an article based on the logical and semantic structure of the article, Technology (P, O, U and R)&mdash; content structures, user agents, assistive technology, hardware and software that support access to digital documents.
</p>

<p>Note: The letters P, O, U and, R stand for perceivable, operable, understandable and robust respectively.
</p>

<p>The base class, barriers, identifies areas where barriers can occur.  The sub-classes are groups of task-specific barrier families. Note a barrier for one set of functional limitations may be an assistance for another. Thus, the sub-classes of barriers are defined in neutral terms like legibility instead of bias terms like "poor legibility" because a presentation that promotes legibility for one group may inhibit legibility for another.
</p>

<h3> Top Level Association Classes:
</h3>

<p>Functional limits are connected to all other base classes. The association classes are: diagnosis-limits &mdash;limits that follow from a diagnosis; effect-limits &mdash; limits attributable to physical conditions; limit-barriers: barriers induced by limits. </p>

<p>The diagnosis and physical-conditions are connected by two associations. The causes-condition class associates a diagnosis with the set of physical-conditions implied by the diagnosis. For example, toxoplasmosis (a diagnosed disease) may cause scotoma (blind spots). The is-diagnosis connects the physical-condition class to the diagnosis class as follows: Some physical-conditions are a diagnosis in their own right. The is-diagnosis links a physical- condition instance with its instance representative in the diagnosis class.</p>

<h2>Data Dictionary of the Top Level Class Structure</h2>

<h3>Base Classes:</h3>

<ul>
    <li>functional-limitation&mdash; Life restrictions imposed by low vision diagnoses and physical conditions.</li>
    <li>diagnosis&mdash; A formal classification of a cause for low vision.</li>
    <li>physical-condition&mdash; physical and behavioral manifestations of low vision diagnoses.</li>
    <li>barrier&mdash; an IT configurations that inhibit access for users with low vision. </li>
</ul>

<h3>Associative Classes:
</h3>

<ul>
    <li>diagnosis-limit &mdash; connects a diagnosis to the functional limits induced by the disease diagnosis. Example: Chorioretinitis induces blurred vision. This is a many to many association.</li>
    <li>condition-limit &mdash; connects physical-conditions to the limits they induce. Example: nystagmus induces reduced acuity.  This is a many to many association.</li>
    <li>causes-condition &mdash; connects diagnoses to physical conditions associated with the disease. Example: Albinism causes photophobia. This association is many to many.</li>
    <li>is-diagnosis &mdash;  connects physical-conditions that are in themselves diagnoses to their instance in the diagnosis class. This association is one to one.</li>
</ul>

<a href="http://www.nosetothepage.org/WCAG/LVTF/Ontology/images/TopER.png">Top level Class Diagram for Low Vision Ontology</a>

<h2>Example: Macular Degeneration (MD)</h2>

<h3>Part A: Computing Functional Limits</h3>

<p>Macular degeneration is a medical diagnosis.  It has associated functional limits and physical conditions.  The functional limits that are directly associated with MD are: acuity (central, reduced), fatigue (visual), color vision (reduced), distortion (blurred vision), distortion (general), fluctuation (slow recovery from changes in light), and sensitivity (to contrast, reduced).</p>

<p>The one physical condition associated directly with MD is:  central scomatas (blind spots)</p>

<p>Scotoma in the central area causes the functional limits directly: acuity (central, reduced), color vision (reduced), field loss (central).</p>

<p>Scotoma is itself a diagnosis and has the physical condition photophobia associated.</p>

<p>Photophobia is associated with the limitations: sensitivity (to light), sensitivity (to glare), discomfort (eye pain), discomfort (squinting), discomfort (closing eyes) and discomfort (headache), and fatigue (eye). There are no additional conditions.</p>

<p>Thus the functional limits associated with macular degeneration are: acuity (central, reduced), color vision (reduced), fatigue (visual), field loss (central), discomfort (eye pain), discomfort (squinting), discomfort (closing eyes), discomfort  (headache), distortion (blurred vision), distortion (general), fluctuation (slow recovery from changes in light), sensitivity (to contrast, reduced),  sensitivity (to light), sensitivity (to glare), and fatigue (eye).</p>

<p>For those familiar with algorithms, this is a standard closure algorithm.</p>

<h3>Expected Barriers for MD</h3>

<p>How are functional-limits and barriers related? This relation is less algorithmic. An analysis of barriers follows:</p>

<ul>
    <li>Legibility for the user is impacted by acuity, color vision, sensitivity (to contrast and light), and distortion (blurred vision). Reduced acuity requires font resizing and a clear font to identify letters. Reduced contrast sensitivity requires increased contrast for legibility. Increased sensitivity to light requires low enough light to avoid glare. Blurred vision requires enough space between letters to identify letters and enough space between words to partition words. It is clear that some of these constraints are contradictory, so balance is required for best legibility.</li>
    <li>Readability in a given environment is impacted by field-loss, discomfort (closed eyes, squinting, eye pain and headaches), eye fatigue, sensitivity (to light and glare) and contrast sensitivity. There two primary goals to be met. Given an environment where maximum legibility is achieved, the reader must minimize errors and fatigue to finish and comprehend articles. Many errors occur when moving through text. Movement within the page is a source of errors. The task must be as easy or easier as it is for normal readers. There are three types of movement: tracking, scanning and scrolling. Tracking is the movement along lines. It is impacted by field loss because readers must "look around" blind spots to see lines and follow lines. Often lacking the full use of the most sensitive part of the retina, the fovea, readers employ "eccentric reading" a practice that enables word recognition but hinders tracking and causes errors. Tracking can be improved with larger line spacing. Field loss also hides content, this inhibits scanning. Other methods like automated search must be used. The extreme discomfort caused by photophobia must be combated with control of light and color.  This may require larger font size with lower contrast as a solution to the legibility problem. The final navigation issue is scrolling, the movement from one viewport to an adjacent viewport of the document.  Normal readers only need to scroll in one direction to read, but enlarged print for legibility can force scrolling in two directions in order to scroll to all views of the document.  Multidirectional scrolling is well known to cause errors and fatigue.</li>
    <li>Layout and Viewport Capacity are impacted primarily by acuity. Reduced acuity requires larger objects and ways to find them. Multi-column formats do not fit. Word wrapping is required. Menu bars, toolbars and panels consume too much space and must be linearized. A 24 inch viewport must be treated like it is a 4 inch mobile viewport.
    </li>
    <li>Way-Finding, the location of items within a page, needs to accommodate reduced acuity, field loss and blurred vision. The reorganized page to accommodate reduced viewport capacity caused by enlargement must still be searchable and navigable.  Navigation techniques provided by screen readers need to be available for the reader with MD. Visual cues like large headings and italicized text may not be appropriate for visual scanning and may require typographic substitutes.</li>
    <li>Technology and the lack of plays a significant role for people with macular degeneration They need accessibility support for all of their functional limits. There is currently no comprehensive assistive technology that supports the needs to avoid barriers to legibility, readability, layout and viewport capacity, and way-finding. While screen readers assist with way-finding the lack of a reasonable visual interface erases this benefit of the tool. Screen magnifiers provide similar access, but their enlargement method complicates reading, neutralizing assistive value. Access to the style features to create legibility and readability is not available to general users, and the structure of current web content discourages this access.</li>
</ul>

<p>From this analysis of limits we can see that adjustable font size and family, letter, line and word spacing, and column formatting are barrier issues for readers with MD. Multi-directional scrolling is also a barrier issue. These users also need navigation tools to find objects lost in blind spots or because of small viewport capacity. This population also lacks an assistive technology or user agent to remove these barriers in total.</p>

<h2>Conclusions</h2>

<p>Construction and population of this ontology will clarify requirements and solutions for accommodating the access needs for users with low vision. The example of macular degeneration illustrates how this ontology can guide prescriptions for accommodations that would support access to web content for this population. Analysis of a representative selection of diagnoses should enable the Task Force to identify the range of issues required to serve people with low vision.</p>

<h2> Reference</h2>

<p>This ontology is based on the article by the Texas School for the Blind and Visually Impaired (TSBVI). The article links diagnoses to physical conditions and functional limits. The table in the TSBVI article is an invaluable resource, but it does not support automated queries. That would be the contribution of an LVTF ontology.</p>

<a href="http://www.tsbvi.edu/eye-conditions"> Specific Eye Conditions, Corresponding Impact on Vision, And Related Educational Considerations </a>
</body>
</html>
